lib/python/picongpu/extra/utils/picmi_density_reader.py

```diff
@@ -0,0 +1,142 @@
+        if self.test_for_cases(s):
+            if self.debug:
+                print("CASES")
+            return self.eval_cases(s, x, symbol)
+
+        else:
+            s_sym = sympy.parsing.sympy_parser.parse_expr(s)
+            res = s_sym.subs(symbol, x)
+            if self.debug:
+                print("eval:", res)
+            return res
+
+    def eval_cases(self, s, x, symbol):
+        """handle c++ cases o form cond ? case1 ? case 2
+        s ... string code
+        x ... float value to evalute
+        symbol ... symbol to replace in string s
+        """
+        if self.debug:
+            print("-->", s)
+        s = self.clean_substring(s)
+        if self.debug:
+            print("==>", s)
+        index_first_q = s.find("?")
+        index_first_c = s.find(":")  # this assumes that there are no cases in the first branch
+
+        condition = s[0:index_first_q]
+        case1 = s[index_first_q + 1 : index_first_c]
+        case2 = s[index_first_c + 1 :]
+
+        if self.debug:
+            print("if")
+            print(condition)
+            print("do")
+            print(case1)
+            print("else")
+            print(case2)
+            print("fi")
+
+        if self.inner_evaluate(condition, x, symbol):
+            return self.inner_evaluate(case1, x, symbol)
+        else:
+            return self.inner_evaluate(case2, x, symbol)
+
+    def initial_clean(self, s):
+        """initially clean string from c++/picongpu methods
+        making it readably for sympy
+        s ... string
+        """
+        if self.debug:
+            print("clean before:", s)
+        s = s.replace("^", "**")
+        s = s.replace("pmacc::math::exp", "exp")
+        s = s.replace("pmacc::math::pow", "pow")
+        if self.debug:
+            print("clean after:", s)
+        return s
+
+    def clean_substring(self, s):
+        """clean any substrings from cases from paraneties and whitespace
+        s ... string
+        """
+        while s[0].isspace():
+            s = s[1:]
+
+        while s[-1].isspace():
+            s = s[:-1]
+
+        if s[0] == "(" and s[-1] == ")":
+            s = s[1:-1]
+        return s
+
+
+if __name__ == "__main__":
+    # load pypicongpu.json, convert json to dict and extract equation for density
+    file = open("pypicongpu.json")
+    sim_dict = json.load(file)
+    density_fct_str = sim_dict["species_initmanager"]["operations"]["simple_density"][0]["profile"]["data"][
+        "function_body"
+    ]
+
+    # create cpp_fct_reader class for later evaluation
+    reader = cpp_fct_reader(density_fct_str)
+
+    # define positions where to evaluate the density
+    x_array = np.linspace(0.0, 5.0e-3, 1000)
+    n_array = np.zeros_like(x_array)
+
+    # evalute density
+    for i, x in enumerate(x_array):
+        n_array[i] = reader.evaluate(x)
+
+    # plot density
+    plt.plot(x_array, n_array)
+    plt.xlabel(r"$y \, \mathrm{[m]}$")
+    plt.xlabel(r"$n \, \mathrm{[m^-3]}$")
```

Comment:
I assume this is y label
```suggestion
    plt.ylabel(r"$n \, \mathrm{[m^-3]}$")
```

Reply:
great catch - fixed